extension/lib/bigcommerce/RequestRepository.js

```diff
@@ -0,0 +1,71 @@
+const Logger = require('./Logger')
+
+class BigCommerceRequestRepository {
+  /**
+   * @param {BigCommerce} client Api V3 client
+   * @param {Logger} logger context.log
+   */
+  constructor (client, logger) {
+    this.client = client
+    this.logger = logger
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  get (path) {
+    return this.request('get', path)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  post (path, data) {
+    return this.request('post', path, data)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  put (path, data) {
+    return this.request('put', path, data)
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  del (path) {
+    return this.request('delete', path)
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} path
+   * @param {Object} data
```

Comment:
Seems like data is optional as you did not pass anything in the `del` method
```suggestion
   * @param {Object} [data]
```